src/HTTPRequest.ts

```diff
@@ -0,0 +1,98 @@
+import * as http from 'http';
+import * as https from 'https';
+import {Digest} from './Digest';
+
+interface Resp {
+    resp: http.IncomingMessage;
+    data?: string;
+}
+
+export interface HttpRequestOptions {
+    method?: string,
+    protocol?: string,
+    host: string,
+    port: number,
+    path: string,
+    auth: string,
+    timeout: number,
+    headers?: {
+        'Content-Type'?: string
+    }
+}
+
+export async function httpRequest(options: HttpRequestOptions, postData?: string, noWaitForData = false) {
```

Comment:
nebylo by zas lepsi to zabalit do clasy a definovat private funkce co nejsou potreba byt viditelne?

Reply:
Bylo nejde to.

Reply:
-